bundles/org.openhab.binding.fronius/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -196,6 +199,30 @@
 		<description>Battery Power ( + charge, - discharge )</description>
 		<state pattern="%.2f W" readOnly="true"></state>
 	</channel-type>
+	<channel-type id="pPv">
+		<item-type>Number</item-type>
+		<label>Solar plant power</label>
+		<description>Current solar plant power</description>
+		<state pattern="%.2f W" readOnly="true"></state>
```

Comment:
```suggestion
		<state pattern="%.2f %unit%" readOnly="true"></state>
```

Reply:
I tried the thing with the %unit% but got some strange behavior. 

thing-types.xml looks like
```xml
	<channel-type id="inverter1Soc">
		<item-type>Number:Dimensionless</item-type>
		<label>Inverter 1 State of Charge</label>
		<description>Inverter 1 State of Charge</description>
		<state pattern="%.1f %unit%" readOnly="true"></state>
	</channel-type>
```

Code looks like (getSoc() returns double)
```java
new QuantityType<>(inverters.get(number).getSoc(), Units.PERCENT);
```

Debugger
```java
NumberQuantity = "91.5999984741211 %"
value = 91.5999984741211
unit = "%"
scale = "RELATIVE"
```

Output switching like this
```log
[INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'FroniusSymoInverter_Inverter1StateofCharge' changed from 950999984741211 % to 95 %
[INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'FroniusSymoInverter_Inverter1StateofCharge' changed from 95 % to 949000015258789 %
[INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'FroniusSymoInverter_Inverter1StateofCharge' changed from 9169999694824219 % to 915999984741211 %
```

What do i miss?

Reply:
Are you sure your Item configuration is correct? Seems like the Item is updated from two different sources.

Reply:
Some more investigation shows the following behavior.

i18n.config
```
:org.apache.felix.configadmin.revision:=L"5"
language="de"
measurementSystem="SI"
service.pid="org.openhab.i18n"
timezone="Europe/Berlin"
```

server locale
```bash
LANG=de_DE.UTF-8
LC_CTYPE="de_DE.UTF-8"
LC_NUMERIC="de_DE.UTF-8"
LC_TIME="de_DE.UTF-8"
LC_COLLATE="de_DE.UTF-8"
LC_MONETARY="de_DE.UTF-8"
LC_MESSAGES="de_DE.UTF-8"
LC_PAPER="de_DE.UTF-8"
LC_NAME="de_DE.UTF-8"
LC_ADDRESS="de_DE.UTF-8"
LC_TELEPHONE="de_DE.UTF-8"
LC_MEASUREMENT="de_DE.UTF-8"
LC_IDENTIFICATION="de_DE.UTF-8"
LC_ALL=
```
Running the openhab server with these settings will result in messed up values like "19563232421875,00" instead of "1956,32".

Running the server with the ```LC_CTYPE``` set to ```en_GB.UTF-8``` will result in the correct values "1956.32" but in the wrong number format (point instead of comma).

```bash
bash> locale
LANG=de_DE.UTF-8
LC_CTYPE="de_DE.UTF-8"
LC_NUMERIC="de_DE.UTF-8"
LC_TIME="de_DE.UTF-8"
LC_COLLATE="de_DE.UTF-8"
LC_MONETARY="de_DE.UTF-8"
LC_MESSAGES="de_DE.UTF-8"
LC_PAPER="de_DE.UTF-8"
LC_NAME="de_DE.UTF-8"
LC_ADDRESS="de_DE.UTF-8"
LC_TELEPHONE="de_DE.UTF-8"
LC_MEASUREMENT="de_DE.UTF-8"
LC_IDENTIFICATION="de_DE.UTF-8"
LC_ALL=

bash> LC_CTYPE=en_GB.UTF-8 ./start_debug.sh
```

So it looks like an issue with the localization of "QuantityType". I think the whole localization should respect the browser languague instead of the server locale.

Reply:
I can hardly belive this behavior is caused by the locale setting. Maybe it's only a symptom. Did you try this with a fresh installation of OH or on another system?

Reply:
With 3.1.0-RC1 the localization issue is gone. See latest patch 76d8b02 .